Mathlib/Data/Real/EReal.lean

```diff
@@ -1227,6 +1525,95 @@ lemma left_distrib_of_nonneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
   nth_rewrite 1 [EReal.mul_comm]; nth_rewrite 2 [EReal.mul_comm]; nth_rewrite 3 [EReal.mul_comm]
   exact right_distrib_of_nonneg ha hb
 
+lemma coe_mul_add_of_nonneg {x : ℝ} (hx_nonneg : 0 ≤ x) (y z : EReal) :
+    x * (y + z) = x * y + x * z := by
```

Comment:
I have been working on this lately. I think it would be better to have something like `{x y z : EReal} (h :  0 ≤ x) (h' : x x ≠ ⊤)`. There are many ways to coerce something into EReal, and your version my be annoying to work with if `x` is e.g. a Nat, a NNReal... On the other hand, if we have a general version on EReal, we can easily deduce any specialized version using various forms on `coe_ne_top`.

Reply:
Same as above.